Git remote branch name

I want to change the git remote branch name form master to main. The documentation says the environment variable to do this is GIT_REMOTE-BRANCH, which is appearently not a valid environment variable name.

az container create --file ./deployment.yaml  
(InvalidContainerEnvironmentVariable) The environment variable name in container 'xx' of container group 'xx' is invalid. A valid environment variable name must start with alphabetic character or '_', followed by a string of alphanumeric characters or '_' (e.g. 'my_name',  or 'MY_NAME',  or 'MyName').

I also tried GIT_REMOTE_BRANCH, however it seems that it does not get pulled into the program.

Just tried but now I get this:

2021-09-24 12:22:50.594  INFO 22 --- [           main] i.m.dockerosb.DockerOsbApplicationKt     : Starting DockerOsbApplicationKt on wk-caas-ded7bb0f662e4f94a7802194a6980f1d-6a76c5de60f2c6f4c46e47 with PID 22 (/app/generic-osb-api-1.0.0.jar started by root in /app)
2021-09-24 12:22:50.658  INFO 22 --- [           main] i.m.dockerosb.DockerOsbApplicationKt     : No active profile set, falling back to default profiles: default
2021-09-24 12:22:56.688  INFO 22 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8075 (http)
2021-09-24 12:22:56.763  INFO 22 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2021-09-24 12:22:56.764  INFO 22 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.36]
2021-09-24 12:22:56.981  INFO 22 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2021-09-24 12:22:56.982  INFO 22 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 6117 ms
2021-09-24 12:23:04.301  INFO 22 --- [           main] i.m.d.persistence.GitHandlerService      : Branch main does not exist locally. Creating it.
2021-09-24 12:23:04.359  WARN 22 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'genericCatalogService' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/service/GenericCatalogService.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gitOperationContextFactory' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/persistence/GitOperationContextFactory.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'gitHandlerService' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/persistence/GitHandlerService.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.meshcloud.dockerosb.persistence.GitHandlerService]: Constructor threw exception; nested exception is org.eclipse.jgit.api.errors.RefNotFoundException: Ref origin/main cannot be resolved
2021-09-24 12:23:04.372  INFO 22 --- [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2021-09-24 12:23:04.393  INFO 22 --- [           main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-09-24 12:23:04.462 ERROR 22 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'genericCatalogService' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/service/GenericCatalogService.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gitOperationContextFactory' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/persistence/GitOperationContextFactory.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'gitHandlerService' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/persistence/GitHandlerService.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.meshcloud.dockerosb.persistence.GitHandlerService]: Constructor threw exception; nested exception is org.eclipse.jgit.api.errors.RefNotFoundException: Ref origin/main cannot be resolved
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:893) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:879) ~[spring-context-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:551) ~[spring-context-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:143) ~[spring-boot-2.3.1.RELEASE.jar!/:2.3.1.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:758) ~[spring-boot-2.3.1.RELEASE.jar!/:2.3.1.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:750) ~[spring-boot-2.3.1.RELEASE.jar!/:2.3.1.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397) ~[spring-boot-2.3.1.RELEASE.jar!/:2.3.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-2.3.1.RELEASE.jar!/:2.3.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1237) ~[spring-boot-2.3.1.RELEASE.jar!/:2.3.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) ~[spring-boot-2.3.1.RELEASE.jar!/:2.3.1.RELEASE]
	at io.meshcloud.dockerosb.DockerOsbApplicationKt.main(DockerOsbApplication.kt:15) ~[classes!/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[generic-osb-api-1.0.0.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:109) ~[generic-osb-api-1.0.0.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[generic-osb-api-1.0.0.jar:na]
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88) ~[generic-osb-api-1.0.0.jar:na]
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'gitOperationContextFactory' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/persistence/GitOperationContextFactory.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'gitHandlerService' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/persistence/GitHandlerService.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.meshcloud.dockerosb.persistence.GitHandlerService]: Constructor threw exception; nested exception is org.eclipse.jgit.api.errors.RefNotFoundException: Ref origin/main cannot be resolved
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:797) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:227) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	... 28 common frames omitted
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'gitHandlerService' defined in URL [jar:file:/app/generic-osb-api-1.0.0.jar!/BOOT-INF/classes!/io/meshcloud/dockerosb/persistence/GitHandlerService.class]: Bean instantiation via constructor failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.meshcloud.dockerosb.persistence.GitHandlerService]: Constructor threw exception; nested exception is org.eclipse.jgit.api.errors.RefNotFoundException: Ref origin/main cannot be resolved
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:313) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:294) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1358) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1204) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:226) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1304) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1224) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:884) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:788) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	... 42 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.meshcloud.dockerosb.persistence.GitHandlerService]: Constructor threw exception; nested exception is org.eclipse.jgit.api.errors.RefNotFoundException: Ref origin/main cannot be resolved
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:217) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:117) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:309) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	... 56 common frames omitted
Caused by: org.eclipse.jgit.api.errors.RefNotFoundException: Ref origin/main cannot be resolved
	at org.eclipse.jgit.api.CreateBranchCommand.getStartPointObjectId(CreateBranchCommand.java:246) ~[org.eclipse.jgit-5.11.0.202103091610-r.jar!/:5.11.0.202103091610-r]
	at org.eclipse.jgit.api.CreateBranchCommand.call(CreateBranchCommand.java:99) ~[org.eclipse.jgit-5.11.0.202103091610-r.jar!/:5.11.0.202103091610-r]
	at org.eclipse.jgit.api.CheckoutCommand.call(CheckoutCommand.java:194) ~[org.eclipse.jgit-5.11.0.202103091610-r.jar!/:5.11.0.202103091610-r]
	at io.meshcloud.dockerosb.persistence.GitHandlerService$Companion.switchToBranchAndCreateIfMissing(GitHandlerService.kt:313) ~[classes!/:na]
	at io.meshcloud.dockerosb.persistence.GitHandlerService$Companion.initGit(GitHandlerService.kt:276) ~[classes!/:na]
	at io.meshcloud.dockerosb.persistence.GitHandlerService.<init>(GitHandlerService.kt:35) ~[classes!/:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source) ~[na:na]
	at kotlin.reflect.jvm.internal.calls.CallerImpl$Constructor.call(CallerImpl.kt:41) ~[kotlin-reflect-1.5.0.jar!/:1.5.0-release-749 (1.5.0)]
	at kotlin.reflect.jvm.internal.KCallableImpl.call(KCallableImpl.kt:108) ~[kotlin-reflect-1.5.0.jar!/:1.5.0-release-749 (1.5.0)]
	at kotlin.reflect.jvm.internal.KCallableImpl.callDefaultMethod$kotlin_reflection(KCallableImpl.kt:159) ~[kotlin-reflect-1.5.0.jar!/:1.5.0-release-749 (1.5.0)]
	at kotlin.reflect.jvm.internal.KCallableImpl.callBy(KCallableImpl.kt:112) ~[kotlin-reflect-1.5.0.jar!/:1.5.0-release-749 (1.5.0)]
	at org.springframework.beans.BeanUtils$KotlinDelegate.instantiateClass(BeanUtils.java:798) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:189) ~[spring-beans-5.2.7.RELEASE.jar!/:5.2.7.RELEASE]
	... 58 common frames omitted

--- a/src/main/kotlin/io/meshcloud/dockerosb/persistence/GitHandlerService.kt
+++ b/src/main/kotlin/io/meshcloud/dockerosb/persistence/GitHandlerService.kt
@@ -6,6 +6,7 @@ import mu.KotlinLogging
 import org.eclipse.jgit.api.*
 import org.eclipse.jgit.merge.MergeStrategy
 import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.transport.RefSpec
 import org.eclipse.jgit.transport.SshSessionFactory
 import org.eclipse.jgit.transport.URIish
 import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider
@@ -266,16 +267,22 @@ class GitHandlerService(
 
         gitConfig.remote?.let {
           ensureRemoteIsAdded(git, gitConfig)
-          val pull = git.pull()
 
-          gitConfig.username?.let {
-            pull.setCredentialsProvider(UsernamePasswordCredentialsProvider(gitConfig.username, gitConfig.password))
-          }
+          // fetch remote
+          git.fetch()
+              .apply {
+                refSpecs = listOf(
+                    RefSpec("refs/heads/${gitConfig.remoteBranch}:refs/remotes/origin/${gitConfig.remoteBranch}")
+                )
 
-          pull.call()
+                gitConfig.username?.let {
+                  setCredentialsProvider(UsernamePasswordCredentialsProvider(gitConfig.username, gitConfig.password))
+                }
+              }
+              .call()
 
+          // checkout a branch
           switchToBranchAndCreateIfMissing(git, gitConfig.remoteBranch)
-
         }
 
         return git
@@ -296,12 +303,16 @@ class GitHandlerService(
     }
 
     private fun ensureRemoteIsAdded(git: Git, gitConfig: GitConfig) {
-      if (git.remoteList().call().isEmpty()) {
-        val remoteAddCommand = git.remoteAdd()
-        remoteAddCommand.setName("origin")
-        remoteAddCommand.setUri(URIish(gitConfig.remote))
-        remoteAddCommand.call()
+      if (git.remoteList().call().isNotEmpty()) {
+        return
       }
+
+      git.remoteAdd()
+          .apply {
+            setName("origin")
+            setUri(URIish(gitConfig.remote))
+          }
+          .call()
     }
 
     private fun switchToBranchAndCreateIfMissing(git: Git, branchName: String) {

--- a/src/test/kotlin/io/meshcloud/dockerosb/RemoteGitFixture.kt
+++ b/src/test/kotlin/io/meshcloud/dockerosb/RemoteGitFixture.kt
@@ -6,11 +6,13 @@ import java.io.File
 import java.nio.file.Path
 
 class RemoteGitFixture(
-    private val repositoryRootPath: String
+    private val repositoryRootPath: String,
+    branch: String = "master"
 ) {
   // init remote git
   private val git = Git.init()
       .setDirectory(File(repositoryRootPath))
+      .setInitialBranch(branch)
       .call()
 
 

--- a/src/test/kotlin/io/meshcloud/dockerosb/ServiceBrokerFixture.kt
+++ b/src/test/kotlin/io/meshcloud/dockerosb/ServiceBrokerFixture.kt
@@ -5,7 +5,6 @@ import io.meshcloud.dockerosb.config.GitConfig
 import io.meshcloud.dockerosb.persistence.GitOperationContextFactory
 import io.meshcloud.dockerosb.persistence.GitHandlerService
 import io.meshcloud.dockerosb.persistence.YamlHandler
-import io.meshcloud.dockerosb.service.GenericCatalogService
 import org.junit.rules.TemporaryFolder
 import java.io.Closeable
 
@@ -31,7 +30,7 @@ class ServiceBrokerFixture(catalogPath: String) : Closeable {
 
   // note: it's important we place the initializer before the constructors below since we need to seed the repo with a
   // catalog before we access it internally
-  val remote = RemoteGitFixture(remoteGitPath).apply {
+  val remote = RemoteGitFixture(remoteGitPath, ).apply {
     initWithCatalog(catalogPath)
   }
 
@@ -40,8 +39,6 @@ class ServiceBrokerFixture(catalogPath: String) : Closeable {
   val gitHandler = GitHandlerService(gitConfig)
 
   val contextFactory = GitOperationContextFactory(gitHandler, yamlHandler)
-
-  val catalogService = GenericCatalogService(contextFactory)
 
   val builder = TestDataBuilder(catalogPath, yamlHandler)
 

--- a/src/test/kotlin/io/meshcloud/dockerosb/service/Git.kt
+++ b/src/test/kotlin/io/meshcloud/dockerosb/service/Git.kt
@@ -1,0 +1,43 @@
+package io.meshcloud.dockerosb.service
+
+import io.meshcloud.dockerosb.RemoteGitFixture
+import io.meshcloud.dockerosb.config.GitConfig
+import io.meshcloud.dockerosb.persistence.GitHandlerService
+import org.eclipse.jgit.revwalk.RevCommit
+import org.junit.Assert
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+
+class GitHandlerServiceTest {
+
+  @Test
+  fun `can use main as a remote branch`() {
+    val tmp = TemporaryFolder().apply {
+      create()
+    }
+
+    try {
+      val localGitPath = tmp.newFolder("git-local").absolutePath
+      val remoteGitPath = tmp.newFolder("git-remote").absolutePath
+
+      val gitConfig = GitConfig(
+          localPath = localGitPath,
+          remote = remoteGitPath,
+          remoteBranch = "main",
+          sshKey = null,
+          username = null,
+          password = null
+      )
+
+      RemoteGitFixture(remoteGitPath, gitConfig.remoteBranch).apply {
+        initWithCatalog("src/test/resources/catalog.yml")
+      }
+
+      val sut = GitHandlerService(gitConfig)
+
+      Assert.assertNotEquals(emptyList<RevCommit>(), sut.getLog());
+    } finally {
+      tmp.delete()
+    }
+  }
+}

